substrate/frame/revive/src/metering/storage.rs

```diff
@@ -76,41 +77,28 @@ pub trait Ext<T: Config> {
 /// It uses [`frame_support::traits::fungible::Mutate`] in order to do accomplish the reserves.
 pub enum ReservingExt {}
 
-/// Used to implement a type state pattern for the meter.
-///
-/// It is sealed and cannot be implemented outside of this module.
-pub trait State: private::Sealed {}
-
-/// State parameter that constitutes a meter that is in its root state.
-#[derive(Default, Debug)]
-pub struct Root;
-
-/// State parameter that constitutes a meter that is in its nested state.
-/// Its value indicates whether the nested meter has its own limit.
-#[derive(Default, Debug)]
-pub struct Nested;
-
-impl State for Root {}
-impl State for Nested {}
-
 /// A type that allows the metering of consumed or freed storage of a single contract call stack.
 #[derive(DefaultNoBound, RuntimeDebugNoBound)]
-pub struct RawMeter<T: Config, E, S: State + Default + Debug> {
+pub struct RawMeter<T: Config, E, S: State> {
 	/// The limit of how much balance this meter is allowed to consume.
-	limit: BalanceOf<T>,
+	pub limit: Option<BalanceOf<T>>,
```

Comment:
this needs to be pub now cause it's use in the math module. 

maybe just to be safe 

```suggestion
	pub(crate) limit: Option<BalanceOf<T>>,
```

same for is_root below

Reply:
I changed this now. But anyway, I don't think that other crates will get direct access to weight and to storage meters, only to the `ResourceMeter`.